feat(cli): warn user if setting autostart on workspace with template-level autostart

mtojek opened the pull request:
## Description

This PR adds a warning message in the CLI when a user sets an autostart schedule for a workspace, and the workspace's template has autostart requirement restrictions (specific days of the week).

## Changes

- **CLI Warning**: When running `coder schedule start <workspace> <schedule>`, the CLI now checks if the template has `AutostartRequirement.DaysOfWeek` restrictions and warns the user
- **Test Coverage**: Added comprehensive tests for the warning functionality

## Example Output

```bash
$ coder schedule start my-workspace 9:30AM Mon-Fri
Warning: your workspace template restricts autostart to the following days: monday, wednesday, friday.
Your workspace may not autostart on days not in this list.
```

## Technical Notes

- The warning is sent to `stderr` (not `stdout`) following CLI conventions
- Warning only appears when setting a schedule (not when setting to "manual")
- In AGPL mode, `AutostartRequirement` always returns all days (enterprise feature), but the warning logic works for both AGPL and Enterprise editions

## Testing

- ✅ All existing schedule tests pass
- ✅ New tests added and passing
- ✅ Linting and formatting pass

Fixes #15619

mtojek committed:
feat(cli): warn user if setting autostart on workspace with template-level autostart

When a user sets a workspace autostart schedule via the CLI, and the
template has autostart requirement restrictions (days of week), warn
the user that their workspace may not autostart on days not allowed
by the template.

This helps users understand why their workspace might not autostart
as expected when template-level restrictions are in place.

Fixes #15619

mtojek committed:
test: remove unnecessary NoWarningWhenNoRequirement test

This test was not applicable in AGPL mode and was always skipped.

johnstcn reviewed (approved):
LGTM modulo one wording suggestion

mtojek committed:
fix: remove double negative in warning message

Use positive wording 'may only autostart on these days' instead of
'may not autostart on days not in this list'.

Co-authored-by: Cian Johnston <cian@coder.com>
